api/proto/evecommon/devmodelcommon.proto

```diff
@@ -19,6 +19,10 @@ enum PhyIoType {
   PhyIoNetWLAN = 5;
   PhyIoNetWWAN = 6;
   PhyIoHDMI = 7;
+  PhyIoLTE = 8;
+  PhyIoNVMEStorage = 9;
+  PhyIoSSDStorage = 10;
```

Comment:
I believe not all SSDs are NVMe SSDs. There are SATA SSDs and SATA HDDs.  So we need three types.

Reply:
What would EVE-OS do differently for a SATA SSD vs a SATA HDD? How would EVE-OS even tell the difference?

The access protocol (NVME vs. SSD) is visible to the kernel and device drivers, and makes a difference for PCI passthrough. But anything downstream of the protocol is invisible.

Reply:
Its mostly for controller to show various storage device types, especially if customer is going to add/replace devices in a pool.

Reply:
Also smartctl should be able to tell NVMe SSD or SATA SSD.

Reply:
So is the assumption that eve-os would use smartctl to fill this in, including spec.sh which creates an rough initial model for the device? 
Will the controller have an opinion about replacing a SATA SSD with a SATA HDD or vice versa? If we have it in the model if would need to reject it since the types of drives would now differ?

This sounds similar to not allowing replacing eth0 with a 10GBASE-T PHY with one which has an optical PHY; why do we need to make that harder by encoding the lower-level information in the type? Seems like it would add the need for more models without any benefits to the users.

Reply:
OK my thought process is based on say a zfs pool has SATA SSDs and the device also has NVMe SSD for PCI passthrough, how do I distinguish between those two device types ? Especially for add/replace drives ?

Reply:
OK if we enforce at model level that all drives have to be of same type then sure we just need two types. I was thinking of devices where there could be mix and match types. Say SATA HDD for eve install, SSD SATA for zpool and NVMe SSD for PCIe passthrough.

Reply:
People can definitely build hardware that way, and then install with EVE on sda, /persist on sdb, and PCI passthrough for the NVME drive(s). But to do the installation they need to specify sda/sdb thus they need to know which drive name to use for what - not whether it is HDD or SDD.

Thus I don't see why we need more than two values; PhyIoNVMEStorage and PhyIoSATAStorage

Reply:
OK my argument is based on the fact that SATA HDD and SATA SSD have different performance characteristics. Since we support pools and add/replace disks in near future, if we do not differentiate  its easy for customers to make mistake and create pools with mixed drives or replace with wrong drives etc resulting into unexpected performance issues.  My gut feeling is if we do not handle now, we will have to revisit this in future. But  if you still insist we have only two types, that's fine I can fix that.

Reply:
10Mbps, 100Mbps, 1G, 10G and 100 Gig Ethernet have quite different performance as well, but we do not have separate type for those. 
One could argue that we should report performance information (for network adapters and storage drives) as additional info, but that doesn't mean it should be part of the type space. That would lead to separate types for 5400 and 7200 rpm rotating disks etc.

Reply:
I see your point. I assume we do not bond or pool together various network interfaces, else that could lead to performance issues. But I completely agree with you that we cannot distinguish same device types with different performance characteristics. I will go ahead fix the commit to use just two device types.